### PR TITLE
refactor: remove legacy collection

### PR DESCRIPTION
**Pull request by xiangce**
- and remove all dependent codes and relevant tests
- see RHINENG-6982

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

*Add your description here*

### CODE REVIEW

**Comment by xiangce** [comment]
This PR requires corresponding modifications to the IQE test. Before this, the IQE test will always fail.

File: insights/client/config.py [hunk 1]
```diff
@@ -18,23 +16,6 @@
 logger = logging.getLogger(__name__)
 
 
-def _core_collect_default():
-    '''
-    Core collection should be disabled by default, unless
-    the RPM version 3.1 or above
```
[the remark this answers is not in the document]
Comment (xiangce) [reply insights/client/config.py:24]: For RHEL 6, or RHEL 7, any installed client version with a version older than `3.0.14`, `core_collect` will be set to True by default, after this change.

**Comment by xiangce** [comment]
Hi @JoySnow - thanks for reviewing. The issue you found was fixed in the latest commit, please have a look again.

**Review by m-horky — APPROVED** [review]
I've been able to run this Core on RHEL 6.10 system with Client 3.0.14. I don't think I have notes.

**Review by JoySnow — APPROVED** [review]
Thanks for the update @xiangce . The `insights-collect` command works well now. 
```
$ insights-collect 
/tmp/insights-xiaoxwan-test321-rhel91.default-20240521082408
```

**Comment by xiangce** [comment]
test me

**Comment by xiangce** [comment]
test me

**Comment by xiangce** [comment]
test me

**Pull request merged**